koku/masu/database/ocp_report_db_accessor.py

```diff
@@ -962,12 +962,23 @@ def populate_cluster_table(self, provider, cluster_id, cluster_alias):
             cluster = clusters.first()
             msg = "fetched entry in reporting_ocp_cluster"
             if not cluster:
-                cluster, created = OCPCluster.objects.get_or_create(
-                    cluster_id=cluster_id, cluster_alias=cluster_alias, provider_id=provider.uuid
-                )
-                msg = f"created entry in reporting_ocp_clusters: {created}"
+                if cluster_id:
+                    cluster, created = OCPCluster.objects.get_or_create(
+                        cluster_id=cluster_id, cluster_alias=cluster_alias, provider_id=provider.uuid
+                    )
+                    msg = f"created entry in reporting_ocp_clusters: {created}"
+                else:
+                    # skip creating new cluster if no cluster_id
+                    LOG.warning(
+                        log_json(
+                            msg="No cluster id provided, skipping creation of new entry in reporting_ocp_clusters",
+                            provider_uuid=provider.uuid,
+                        )
+                    )
+                    return None
```

Comment:
It looks like this cluster needs to exist here:
```
populate_openshift_cluster_information_tables
```

Reply:
Yes you make a good point there, oversight on my end. 
I've updated it to raise an error versus returning `None`

Reply:
So I think this solution is _okay_, I do think there is a much better place for this to go. If we don't have a cluster-id, then we shouldn't be doing _anything_ because the cluster-id is used practically everywhere. We retrieve the cluster-id and alias when we instantiate the `OCPReportParquetSummaryUpdater`. Could we validate that we have the id and alias in that init method?

Reply:
I see. I'll add this check in the `OCPReportParquetSummaryUpdater` init method 

```py
...
self._cluster_id = get_cluster_id_from_provider(self._provider.uuid)
if not self._cluster_id:
    raise ValueError(<error_msg>)
...
```